modules/mod_acl_user_groups/support/acl_user_groups_checks.erl

```diff
@@ -661,31 +652,40 @@ can_rsc_for_collab_ugs(Id, Action, CollabId, CatId, UGs, Context) ->
     AllCollabId = m_rsc:rid(acl_collaboration_group, Context),
     lists:any(
         fun (GId) ->
-            case {
-                can_rsc_for_collab_ug(Id, Action, CollabId, CatId, GId, Context),
-                can_rsc_for_collab_ug(Id, Action, AllCollabId, CatId, GId, Context)
-            } of
-                {false, _AllCollabPermission} -> false;
-                {_CollabPermission, false} -> false;
-                {true, _AllCollabPermission} -> true;
-                {undefined, undefined} -> false;
-                {undefined, AllCollabPermission} -> AllCollabPermission
+            case can_rsc_for_collab_ug(Id, Action, CollabId, CatId, GId, Context) of
+                false ->
+                    % Deny rule: no permission to see the resource of the collab group itself
+                    false;
+                MaybePermission ->
+                    case can_rsc_for_collab_ug(Id, Action, AllCollabId, CatId, GId, Context) of
+                        false ->
+                            % Deny rule: No permission to see this content inside collab groups I am member of
+                            false;
+                        true ->
+                            % Allow rule: Permission to see this content inside collab groups I am member of
+                            true;
+                        undefined when MaybePermission =:= true ->
+                            % No rule set, use rights on the collab group resource
+                            true;
+                        undefined when MaybePermission =:= undefined ->
+                            % No rule set, use rights on the collab group resource
+                            false
+                    end
             end
         end,
-        UGs
-    ).
+        UGs).
 
 can_rsc_for_collab_ug(Id, Action, CollabId, CatId, GId, Context) ->
-    case {
-        mod_acl_user_groups:await_lookup({CollabId, {CatId, Action, false}, GId}, Context),
-        is_owner(Id, Context),
-        mod_acl_user_groups:await_lookup({CollabId, {CatId, Action, true}, GId}, Context)
-    } of
-        {Permission, false, _} -> Permission;
-        {undefined, true, undefined} -> undefined;
-        {undefined, true, OwnerPermission} -> OwnerPermission;
-        {Permission, true, undefined} -> Permission;
-        {Permission, true, OwnerPermission} -> Permission orelse OwnerPermission
+    case mod_acl_user_groups:await_lookup({CollabId, {CatId, Action, false}, GId}, Context) of
+        true ->
+            true;
+        FalseOrUndefined ->
+            case is_owner(Id, Context) of
+                true ->
+                    mod_acl_user_groups:await_lookup({CollabId, {CatId, Action, true}, GId}, Context);
```

Comment:
I think this is different from before:
await_lookup(..false..) = `false`
is_owner = `true`
await_lookup(..true..) = `undefined`
=> **false** vs now: **undefined**